src/containers/sign-on/SignOnProvider.tsx

```diff
@@ -237,7 +237,8 @@ export class SignOnProvider extends Component<SignOnProps, SignOnState> {
     return () => {
       const { email } = this.props.fields
       if (page === Pages.PASSWORD) {
-        this.props.recordCompleteEmail(email.value)
+        this.props.onEmailSubmitted(email.value)
```

Comment:
This breaks the existing pattern and requires the saga to handle recording the event, which I don't like. But we do need to ensure we check the email prior to attempting to move pages...